xlators/cluster/afr/src/afr-common.c

```diff
@@ -4195,7 +4151,38 @@ __afr_fd_ctx_set(xlator_t *this, fd_t *fd)
 out:
     if (ret && fd_ctx)
         _afr_cleanup_fd_ctx(this, fd_ctx);
-    return ret;
+    return fd_ctx;
+}
+
+afr_fd_ctx_t *
```

Comment:
```suggestion
static afr_fd_ctx_t *
```

Reply:
It can be done but I found that it  is not just about putting "static" keyword.
This will require a lot of rearrangement of code in this file which 
is not related to main issue and could cause possible new issues.
So keeping it as it is.

Reply:
I don't see how it can introduce new issues. Perhaps I'm missing something. The whole point was to optimize.

Reply:
So, if we make these function static, it gives compile time "warning" as the other non static definition is found in afr.h.
If I remove those definition from afr.h then the arrangements of function definition on this file comes into picture. If func1 calls func2, func2 should be defined above func1.
Now, if I start rearranging these function it will require a lot of time to do it correctly as func1 is calling funct2 and funct3 and so on and they could be defined anywhere in this file, which was working fine as we have defined prototype in afr.h and included it.
Also, there are many non static functions in this file which could be static if we re factor and rearrange this file. 
So, I think we should have an issue opened to do only for these kind of changes. Otherwise we, developer and reviewer, will lose focus on what actually was supposed to be done for this issue and that certainly can cause bug in the patch.

Reply:
You are right of course. This is not a one-liner change, even if it's a trivial change. I've done it here - https://review.gluster.org/#/c/glusterfs/+/23792/ and obviously no one took it as it was a lot of moving code around :-/

I'm just unsure if we don't use static we actually get some benefit from this change.

Reply:
@mykaul  even I am not sure about this. This is always debatable and people have different preferences.

Reply:
ACK. It's clearly best practices in other projects, for both performance and readability, but not in Gluster. Not sure why but I haven't had much success convincing other community members to set this as a standard.
Overall (IMHO!) if the community doesn't see the value in such changes, let's not invest in them.

Reply:
@mykaul @pranithk @itisravi Review approval please if you don't see any other issue with this patch.